File: BussinesLogic/Servicios/ServicioEtapasTaller.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using Dominio.Interfaces.Aplicacion.EtapasTaller;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioEtapasTaller : IServicioEtapasTaller
    {
        private readonly IRepositorioEtapasTaller repositorioEtapasTaller;

        public ServicioEtapasTaller(IRepositorioEtapasTaller repositorioEtapasTaller)
        {
            this.repositorioEtapasTaller = repositorioEtapasTaller;
        }
        public async Task<DtoRespuestaSP> InsertarEtapasTaller(EntitiEtapasTaller entitiEtapasTaller)
        {
            return await this.repositorioEtapasTaller.InsertarEtapasTaller(entitiEtapasTaller);
        }
        public async Task<DtoRespuestaSP> ActualizarEtapasTaller(EntitiEtapasTaller entitiEtapasTaller)
        {
            return await this.repositorioEtapasTaller.ActualizarEtapasTaller(entitiEtapasTaller);
        }
        public async Task<DtoRespuestaSP> EliminarEtapasTaller(int idEtapaTaller)
        {
            return await this.repositorioEtapasTaller.EliminarEtapasTaller(idEtapaTaller);
        }
        public async Task<DtoEtapasTaller> ObtenerEtapasTallerPorID(int idEtapaTaller)
        {
            return await this.repositorioEtapasTaller.ObtenerEtapasTallerPorID(idEtapaTaller);
        }

        public async Task<List<DtoEtapasTaller>> ObtenerEtapasTaller()
        {
            return await this.repositorioEtapasTaller.ObtenerEtapasTaller();
        }
    }
}

File: CX/ClientApp/src/servicios/ServicioEtapasTaller.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "etapastaller";

export const AgregarEtapasTaller = async (data) => {
    const url = `${controlador}/insertaretapastaller`;
    return await ProcesarDatosApi('POST', url, data);
}

export const ActualizarEtapasTaller  = async (data) => {
    const url = `${controlador}/actualizaretapastaller`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const InactivarEtapasTaller  = async (id) => {
    const url = `${controlador}/eliminaretapastaller?idEtapaTaller=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}

export const ObtenerEtapasTallerPorId = async (id) => {
    const url = `${controlador}/obteneretapastallerporid/${id}`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerEtapasTaller  = async () => {
    const url = `${controlador}/Obteneretapastaller`;
    return await ProcesarDatosApi('GET', url);
}
